app/models/concerns/profile_searching.rb

```diff
@@ -35,8 +35,22 @@ module ProfileSearching
       where(external: external)
     }
     scope :has_test_results, lambda { |has_test_results = true|
-      test_results = TestResult.select(:profile_id).distinct
-      has_test_results && where(id: test_results) || where.not(id: test_results)
+      # the use of default scope is to bypass Pundit
+      # and avoid the situation with bind parameters
+      # that must have a different order in the resulting query
+      # of a scoped_search
+      policy_test_results = default_scoped.joins(policy_object: :test_results)
```

Comment:
```suggestion
      policy_test_results = policy_object.test_results
```

Reply:
`policy_object` is on an instance, not on a class:
```
[1] pry(main)> Profile.policy_object
NoMethodError: undefined method `policy_object' for Profile (call 'Profile.connection' to establish a connection):Class
```

Reply:
Oh, right. Good point! lol